encodings/datetime-parts/src/compress.rs

```diff
@@ -2,15 +2,19 @@ use vortex::array::datetime::{TemporalArray, TimeUnit};
 use vortex::array::primitive::PrimitiveArray;
 use vortex::compute::unary::try_cast;
 use vortex::{Array, IntoArray, IntoArrayVariant};
-use vortex_dtype::PType;
+use vortex_dtype::{DType, Nullability, PType};
 use vortex_error::{vortex_bail, VortexResult};
 
 /// Compress a `TemporalArray` into day, second, and subsecond components.
 ///
 /// Splitting the components by granularity creates more small values, which enables better
 /// cascading compression.
 pub fn compress_temporal(array: TemporalArray) -> VortexResult<(Array, Array, Array)> {
-    let timestamps = try_cast(&array.temporal_values(), PType::I64.into())?.into_primitive()?;
+    let timestamps = try_cast(
+        &array.temporal_values(),
+        &DType::Primitive(PType::I64, Nullability::Nullable),
```

Comment:
`CastFn` has poorly defined semantics for what the DType of the resulting array is, which is causing panic when building new chunkedarray. We should probably choose a semantic and document it as part of #260

Reply:
this is no longer a problem now that we're not pushing chunking internally, so i'm removing this